pages/index.js

```diff
@@ -10,22 +10,66 @@ const Subline = styled.h2`
   cursor: default;
 `;
 
+const initialFilter = {
+  time: null,
+  difficulty: null,
+  category: null,
+  priceCategory: null,
+};
+
 export default function HomePage({ projects, favorites, onToggleFavorite }) {
   const [projectsToDisplay, setProjectsToDisplay] = useState(projects);
+  const [filter, setFilter] = useState(initialFilter);
 
-  function handleFilter(filterData) {
-    setProjectsToDisplay(
-      projects.filter(
-        (project) =>
-          project.priceCategory === filterData.priceCategory &&
-          project.difficulty === filterData.difficulty &&
-          filterData.category.includes(project.category.toLowerCase()) &&
-          filterData.time >= project.time
-      )
-    );
+  useEffect(() => {
+    handleFilter(filter);
+  }, [filter]);
+
+  function handleFilter() {
+    const filteredByTime = filter.time
+      ? projects.filter((project) => project.time <= filter.time)
+      : projects;
+
+    const filteredByCategory = filter.category
+      ? filteredByTime.filter((project) => project.category === filter.category)
+      : filteredByTime;
+
+    const filteredByPriceCategory = filter.priceCategory
+      ? filteredByCategory.filter(
+          (project) => project.priceCategory === filter.priceCategory
+        )
+      : filteredByCategory;
+
+    const filteredByDifficulty = filter.difficulty
+      ? filteredByPriceCategory.filter(
+          (project) => project.difficulty === filter.difficulty
+        )
+      : filteredByPriceCategory;
+    setProjectsToDisplay(filteredByDifficulty);
```

Comment:
here you need to return the filtered Projects instead of calling the setter function